low_level_platform/impl/src/lf_rp2040_support.c

```diff
@@ -134,15 +134,17 @@ int _lf_interruptable_sleep_until_locked(environment_t* env, instant_t wakeup_ti
   sem_reset(&_lf_sem_irq_event, 0);
   // create us boot wakeup time
   target = from_us_since_boot((uint64_t)(wakeup_time / 1000));
-  // allow interrupts
-  LF_CRITICAL_SECTION_EXIT(env);
+  // Enable interrupts. NOTE: It would be nice to use the macro LF_CRITICAL_SECTION_EXIT,
```

Comment:
Why would this be nice? Tagging @petervdonovan who introduced the low-level platform API. My sense is that the code change is correct but the comment isn't...

Reply:
I think that this comment is correct. There is a complexity in the build system architecture which prevents this code in the `low_level_platform` module from depending on `util.h`, which is in the `core` module. The `core` module depends on `low_level_platform`, so we prefer that `low_level_platform` does not also depend on `core`.

In my view, this complexity is essential complexity, not accidental complexity, and it is not really specific to CMake.

To make the code easy to understand, we would like for the dependency graph between the different modules to be a DAG; this way it is possible in principle to check the correctness of the most fundamental modules, then check the modules that depend on them, and so forth. If there are cyclic dependencies between modules, then everything in the cycle must be understood in order to understand one part. Therefore, in our preferred approach there is essential complexity in avoiding cyclic dependencies between `core` and `low_level_platform`.

If we want to use that macro in `low_level_platform`, then probably we would need to move the macro out of `core` and into `low_level_platform`. I do not have an opinion about whether such a refactoring is necessary, however.

Reply:
The comment may be correct in the sense that it describes why the macro cannot be used, but I don't think it's useful to express a desire for using it 1. because there is a good reason this isn't possible and 2. because the code written here is functionally equivalent and isn't any less elegant. I suggest to just strike the comment.
```suggestion
  // Enable interrupts.
```